src/internal/testutil/identity.go

```diff
@@ -22,28 +23,41 @@ import (
 const DexMockConnectorEmail = `kilgore@kilgore.trout`
 
 // OIDCOIDCConfig is an auth config which can be used to connect to the identity service in tests
-func OIDCOIDCConfig() *auth.OIDCConfig {
+func OIDCOIDCConfig(host, issuerPort, redirectPort string, local bool) *auth.OIDCConfig {
+
 	return &auth.OIDCConfig{
-		Issuer:          "http://pachd:1658/dex",
+		Issuer:          "http://" + host + ":" + issuerPort + "/dex",
 		ClientID:        "pachyderm",
 		ClientSecret:    "notsecret",
-		RedirectURI:     "http://pachd:1657/authorization-code/callback",
-		LocalhostIssuer: true,
+		RedirectURI:     "http://" + host + ":" + redirectPort + "/authorization-code/callback",
+		LocalhostIssuer: local,
 		Scopes:          auth.DefaultOIDCScopes,
 	}
 }
 
 // ConfigureOIDCProvider configures the identity service and the auth service to
 // use a mock connector.
-func ConfigureOIDCProvider(t *testing.T, c *client.APIClient) error {
+func ConfigureOIDCProvider(t *testing.T, c *client.APIClient, realEnv bool) error {
 	adminClient := AuthenticateClient(t, c, auth.RootUser)
 
 	_, err := adminClient.IdentityAPIClient.DeleteAll(adminClient.Ctx(), &identity.DeleteAllRequest{})
 	require.NoError(t, err)
 
+	issuerHost := "pachd"
+	issuerPort := "1658"
+	redirectPort := "1657"
+	local := true
+
+	if realEnv {
+		issuerHost = c.GetAddress().Host
```

Comment:
what does this value typically evaluate to? I'm wondering if this could work for all cases

Reply:
Leaving this as "pachd" might be okay. I can test that out. But this is the host name generated by server that gets spun up in mockPachd. In my case `127.0.0.1` or localhost. The key is actually the port, since we listen on a randomly generated port which allows us to run multiple tests in parallel.